build.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python
+
+# Keep this file compatible with Python 2 -- it's the first file users will run.
+# Even if we end up requiring a different version for the rest of the project,
+# this file should probably not crash and burn.
+
+import argparse
+import errno
+import itertools
+import os
+import subprocess
+import sys
+import traceback
+
+try: import ctypes  # necessary for Windows
+except ImportError: ctypes = None
+
+pip = None  # Will be lazily imported if needed (lazy because the rest of this script should run even on installations without pip)
+
+StringishTypes = (str, bytes) if sys.version_info[0] >= 3 else (basestring, str, unicode)
+
+def pip_main(*args):
```

Comment:
Let's get rid of this and call pip via subprocess, in a similar way as it is done in the shell scripts (but with proper handling of whitespaces in paths).